packages/mobile/src/hooks/useAppLifecycle.ts

```diff
@@ -0,0 +1,76 @@
+import { useRef } from 'react'
+
+import { useAppState } from '@react-native-community/hooks'
+import { NativeEventEmitter, NativeModules, Platform } from 'react-native'
+import RNRestart from 'react-native-restart'
+import TrackPlayer, { State } from 'react-native-track-player'
+import { useAsync, useEffectOnce } from 'react-use'
+
+const eventEmitter = new NativeEventEmitter(
+  NativeModules.DeviceEventManagerModule
+)
+
+// Time in milliseconds after which we consider the app to be "long backgrounded"
+const BACKGROUND_THRESHOLD = 1000 * 60 * 5 // 5 minutes
+
+const useRestartStaleApp = () => {
+  const backgroundTimeRef = useRef<number | null>(null)
+  const currentAppState = useAppState()
+  const prevAppStateRef = useRef(currentAppState)
+
+  useAsync(async () => {
```

Comment:
If I understand correctly, we're detecting on mount of the app component whether it's been a certain amount of time since we last mounted the component. And if so we're interrupting everything that's going on and force-restarting the app? I'm wondering if there's a more synchronous way to do this so that we don't fire off a bunch of network requests and then restart the app.

I think a layout effect would help, but if we have to make an async call to get player state, I'm not sure we'll avoid doing a bunch of work right before restarting.

Is there a way for us to set a timer of sorts and kill the app after that amount of time, maybe with the ability to check the player state from native Android code?

Reply:
yeah i agree with all of this, let me have a think